pintc/src/pint_parser.lalrpop

```diff
@@ -1238,6 +1247,47 @@ IdentFromToken<Tok>: Ident = {
     }
 }
 
+/////////////////
+// ASM Blocks ///
+/////////////////
+AsmBlockExpr: ExprKey = {
+    <l:@L> "asm" "(" <args:SepList<Expr, ",">> ")" "{" <ops:AsmOp*> "}" <r:@R> => {
+        let span = (context.span_from)(l, r);
+        context.contract.exprs.insert(
+            Expr::AsmBlock {
+                args,
+                ops,
+                span: span.clone(),
+            },
+            Type::Unknown(span),
+        )
+    }
+}
+
+AsmOp: AsmOp = {
+    <l1:@L> <op:Ident> <l2:@L> <arg:"int_lit"?> <r:@R> => {
```

Comment:
It looks like you're not using `l1` here.  And we should have it so parse_int_immediate() can't fail, or just panics.  By parsing the string of the int we should be validating that it's correct before we try to convert it to a number.  Can't fail.

Reply:
Wait, I'm confused. The lexer already checks that the string is an integer. `parse_int_immediate` then checks whether it's a _valid_ int. In this case, we also want to exclude b256s. So, `parse_int_immediate` will basically fail if the int is too wide for an `i64`.

Reply:
I reworded the error message a bit. If the literal is not an integer at all, the parser will emit the usual `expected...found` error.

Reply:
Ahh, sorry, I made assumptions about `parse_int_immediate()` without refreshing myself on what it actually does.  I guess the function name should be `validate_int_immediate()`?

Reply:
Well... it does take a string and output an `Immediate` but the immediate could be a `B256` which we don't want here but do want in other places.